### PR TITLE
Push new version of SHELL Encryption Library.

### PR DESCRIPTION
**Pull request by james-choncholas**
Includes build fixes for tink dependencies.

### CODE REVIEW

**Review by kwlyeo — APPROVED** [review]

File: shell_encryption/galois_key.h [hunk 1]
```diff
@@ -20,6 +20,8 @@
 #include <vector>
 
 #include "absl/strings/str_cat.h"
+
```
[the remark this answers is not in the document]
Comment (james-choncholas) [reply shell_encryption/galois_key.h]: Fixed.

File: shell_encryption/galois_key.h [hunk 2]
```diff
@@ -88,7 +90,7 @@ class GaloisKey {
   // SerializedGaloisKey is (2 * num_parts * dimension) where dimension is the
   // number of digits needed to represent the modulus in base
   // 2^{log_decomposition_modulus}. Crashes for non-valid input parameters.
-  static rlwe::StatusOr<GaloisKey> Deserialize(
+  static  rlwe::StatusOr<GaloisKey> Deserialize(
```
[the remark this answers is not in the document]
Comment (james-choncholas) [reply shell_encryption/galois_key.h]: Fixed.

File: shell_encryption/galois_key.h [hunk 3]
```diff
@@ -49,7 +51,7 @@ class GaloisKey {
   // power of x in the secret key polynomial s(x^substitution_power) that the
   // ciphertext is encrypted with. The prng_seed is used to generate and encode
   // the bottom row of the matrix, which consists of random entries.
-  static rlwe::StatusOr<GaloisKey> Create(
+  static  rlwe::StatusOr<GaloisKey> Create(
```
[the remark this answers is not in the document]
Comment (james-choncholas) [reply shell_encryption/galois_key.h]: Fixed.

File: shell_encryption/int256.h [hunk 4]
```diff
@@ -16,6 +16,7 @@
 #ifndef RLWE_INT256_H_
 #define RLWE_INT256_H_
 
+
```
[the remark this answers is not in the document]
Comment (james-choncholas) [reply shell_encryption/int256.h]: Fixed.

File: shell_encryption/montgomery.cc [hunk 5]
```diff
@@ -14,6 +14,8 @@
 
 #include "shell_encryption/montgomery.h"
 
+
+
```
[the remark this answers is not in the document]
Comment (james-choncholas) [reply shell_encryption/montgomery.cc]: Fixed.

File: shell_encryption/montgomery.cc [hunk 6]
```diff
@@ -472,16 +474,27 @@ MontgomeryInt<T> MontgomeryInt<T>::MultiplicativeInverse(
...
-template struct MontgomeryIntParams<Uint16>;
-template struct MontgomeryIntParams<Uint32>;
-template struct MontgomeryIntParams<Uint64>;
-template struct MontgomeryIntParams<absl::uint128>;
-template class MontgomeryInt<Uint16>;
-template class MontgomeryInt<Uint32>;
-template class MontgomeryInt<Uint64>;
-template class MontgomeryInt<absl::uint128>;
+template struct
+    MontgomeryIntParams<Uint16>;
```
[the remark this answers is not in the document]
Comment (james-choncholas) [reply shell_encryption/montgomery.cc]: Fixed.

File: shell_encryption/montgomery.h [hunk 7]
```diff
@@ -75,7 +77,8 @@ struct BigInt<unsigned __int128> {
 // The parameters necessary for a Montgomery integer. Note that the template
 // parameters ensure that T is an unsigned integral of at least 8 bits.
 template <typename T>
-struct MontgomeryIntParams {
+struct
```
[the remark this answers is not in the document]
Comment (james-choncholas) [reply shell_encryption/montgomery.h]: Fixed.

File: shell_encryption/montgomery.h [hunk 8]
```diff
@@ -223,7 +227,8 @@ MontgomeryIntParams<unsigned __int128>::BarrettReduceBigInt(
 // The underlying integer type T must be unsigned and must not be bool.
 // This class is thread safe.
 template <typename T>
-class ABSL_MUST_USE_RESULT MontgomeryInt {
+class
+    ABSL_MUST_USE_RESULT MontgomeryInt {
```
[the remark this answers is not in the document]
Comment (james-choncholas) [reply shell_encryption/montgomery.h]: Fixed.

File: shell_encryption/montgomery_test.cc [hunk 9]
```diff
@@ -81,7 +83,8 @@ inline unsigned __int128 GetIntMax<unsigned __int128>() {
 #endif
 
 template <typename T>
-class MontgomeryTest : public ::testing::Test {};
+class
+    MontgomeryTest : public ::testing::Test {};
```
[the remark this answers is not in the document]
Comment (james-choncholas) [reply shell_encryption/montgomery_test.cc]: Fixed.

File: shell_encryption/symmetric_encryption.h [hunk 10]
```diff
@@ -761,6 +760,9 @@ class SymmetricRlweKey {
 
   // Accessors.
   unsigned int Len() const { return key_.Len(); }
+  // b/278777783: These accessors should return const references since they are
```
[the remark this answers is not in the document]
Comment (james-choncholas) [reply shell_encryption/symmetric_encryption.h]: Fixed.

File: shell_encryption/relinearization_key.h [hunk 11]
```diff
@@ -57,7 +67,8 @@ namespace rlwe {
 // length (k - 1), where k is the number of parts of the ciphertext it applies
 // to.
 template <typename ModularInt>
-class RelinearizationKey {
+class
```
[the remark this answers is not in the document]
Comment (james-choncholas) [reply shell_encryption/relinearization_key.h]: Fixed.

File: shell_encryption/montgomery.h [hunk 12]
```diff
@@ -502,103 +510,108 @@ class ABSL_MUST_USE_RESULT MontgomeryInt {
...
-                                              const Params* params);
 
   // Equality.
   bool operator==(const MontgomeryInt& that) const { return (n_ == that.n_); }
   bool operator!=(const MontgomeryInt& that) const { return !(*this == that); }
 
   // Modular exponentiation.
-  MontgomeryInt ModExp(Int exponent, const Params* params) const;
+   MontgomeryInt ModExp(Int exponent,
+                                               const Params* params) const;
```
[the remark this answers is not in the document]
Comment (james-choncholas) [reply shell_encryption/montgomery.h:601]: This line break should stay, it exists in google3.

File: shell_encryption/prng/single_thread_chacha_prng.h [hunk 13]
```diff
@@ -52,8 +53,9 @@ class SingleThreadChaChaPrng : public SecurePrng {
   //
   // Fails if the key is not the expected size or on internal cryptographic
   // errors.
-  static rlwe::StatusOr<std::unique_ptr<SingleThreadChaChaPrng>> Create(
-      absl::string_view in_key);
+  static
```
[the remark this answers is not in the document]
Comment (james-choncholas) [reply shell_encryption/prng/single_thread_chacha_prng.h]: Fixed.

**Pull request merged**